src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1086,7 +1086,7 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
         create.sql_security = std::make_shared<ASTSQLSecurity>();
 
     if (create.sql_security)
-        processSQLSecurityOption(getContext(), create.sql_security->as<ASTSQLSecurity &>(), create.attach, create.is_materialized_view);
+        processSQLSecurityOption(getContext(), create.sql_security->as<ASTSQLSecurity &>(), create.is_materialized_view, /* skip_check_permissions= */ is_restore_from_backup || create.attach);
```

Comment:
Please use `LoadingStrictnessLevel` instead of `create.attach`, so it will work correctly for distributed DDL. Also, we can get rid of `is_restore_from_backup` and replace it with `SECONDARY_CREATE`